### DISCUSSION TITLE
Better diagnostics when converting to C++ structs?

### DISCUSSION THREAD

**Discussion by fferri** [danielaparker/jsoncons#442] [Q&A]
When converting `Json` objs to C++ `struct`s (via the various traits macros), and the data doesn't match the schema declared in those macros, we get a "naked" error that doesn't mention the field, e.g. "Not an integer" or "Not a *StructName*: Unable to convert into the provided type", and for structs with lot of fields of the same type, this becomes very hard to debug.

Is there any way to get more detailed error diagnostics?

**Comment by danielaparker** [comment]
This should be addressed on master, the field name will be appended to the message.
